### PR TITLE
[ticket/16159] Wrap post times in html time tag

### PR DESCRIPTION
**Pull request by stevendegroote**
PHPBB3-16159

Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16159

### CODE REVIEW

**Comment by 3D-I** [comment]
IMHO that's wrong.

`$user->format_date()` already formats the date depending on users' choice in UCP.

Why would you want to force a date format for everybody?
Not to mention that this will break BC.

Or am I missing something?

File: phpBB/includes/functions_display.php [hunk 1]
```diff
@@ -547,11 +547,12 @@ function display_forums($root_data = '', $display_moderators = true, $return_mod
 				$last_post_subject = $last_post_subject_truncated = '';
 			}
 			$last_post_time = $user->format_date($row['forum_last_post_time']);
+			$last_post_time_rfc3339 = gmdate("Y-m-d H:i:sP", $row['forum_last_post_time']);
```
Comment (3D-I) [inline phpBB/includes/functions_display.php]: Why?

**Comment by stevendegroote** [comment]
The output format for the user is still configurable, that doesn't change at all.
What changes is that the outputted dates are put in a html <time> tag.

Please see https://tracker.phpbb.com/browse/PHPBB3-16159 for an explanation.

**Comment by 3D-I** [comment]
How can it be congigurable if you are forcing a format?

`gmdate("Y-m-d H:i:sP"`

**Comment by 3D-I** [comment]
@stevendegroote 

I had the opportunity to discuss the subject, sorry for my healthy doubts due to an extemporaneous objective vision. Thank you :+1:

File: phpBB/includes/functions_display.php (hunk 1 above)
Comment (hanakin) [reply phpBB/includes/functions_display.php]: What’s with the rfc3339 stuff? Are you fixing all date times I. The theme to work with the time and datetime setup in this pr?

Comment (stevendegroote) [reply phpBB/includes/functions_display.php]: No no. Just check the ticket description in the issue tracker, and check in the template files where these RFC3339 dates are used. They don't change anything visually, and maintain the ability for users to configure their own date format for output.

Comment (3D-I) [reply phpBB/includes/functions_display.php]: @hanakin 
I had the same impression I have to admit but that's just a wrapper for the sake of search engines, at the end of all.

Why not to use DATE_ATOM instead? That's the same as of DATE_RFC3339.

https://www.php.net/manual/en/class.datetimeinterface.php#datetime.constants.rfc3339

Comment (stevendegroote) [reply phpBB/includes/functions_display.php]: @3D-I You have a point about the format. Maybe we can use some PHP constant. According to MDN documentation for <time>, there are several valid formats. 

I'll recheck the exact format. It looks like the format I'm using is actually not fully RFC3339 (as it's missing the T between date and time now).

Comment (3D-I) [reply phpBB/includes/functions_display.php]: **RFC3339_EXTENDED** is for PHP >= 7 if I correctly recall?

Comment (stevendegroote) [reply phpBB/includes/functions_display.php]: I replaced the date formats with DATE_RFC3339 (available on required PHP versions, and the actual name of the standard). ATOM is a feed standard, which happens to use RFC3339 as its format for dates.

**Comment by marc1706** [comment]
Thanks for your PR. Good job.

**Pull request merged**